src/bashi/filter_software_dependency.py

```diff
@@ -103,5 +113,17 @@ def software_dependency_filter(
                     f"{row[CMAKE].version}",
                 )
                 return False
+    # Rule: d4
+    # all ROCm images are Ubuntu 20.04 based
+    """
+    if UBUNTU in row and row[UBUNTU].version >= pkv.parse("20.04"):
+        for compiler_type in (HOST_COMPILER, DEVICE_COMPILER):
+            if compiler_type in row and row[compiler_type].name == HIPCC:
+                if ALPAKA_ACC_GPU_HIP_ENABLE in row and row[ALPAKA_ACC_GPU_HIP_ENABLE].version != OFF_VER:
+                    return False
+    """
+    if UBUNTU in row and row[UBUNTU].version >= pkv.parse("20.04"):
```

Comment:
```suggestion
    if UBUNTU in row and row[UBUNTU].version < pkv.parse("20.04"):
```
The version check is wrong. It will catch all modern Ubuntu versions